internal/controller/serverset/serverset.go

```diff
@@ -337,21 +338,28 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	// for n times of cr.Spec.Replicas, create a server
 	// for each server, create a volume
 	e.log.Info("Creating a new ServerSet", "name", cr.Name, "replicas", cr.Spec.ForProvider.Replicas)
+	errGroup, ctx := errgroup.WithContext(ctx)
 	for i := 0; i < cr.Spec.ForProvider.Replicas; i++ {
-		volumeVersion, serverVersion, err := getVersionsFromVolumeAndServer(ctx, e.kube, cr.GetName(), i)
-		if err != nil && !errors.Is(err, errNoVolumesFound) {
-			return managed.ExternalCreation{}, err
-		}
-		if err := e.ensureServerAndNicByIndex(ctx, cr, i, serverVersion); err != nil {
-			return managed.ExternalCreation{}, err
-		}
+		errGroup.Go(func() error {
```

Comment:
this also creates bootvolumes in parallel. If we do that, we need to test what is written in the configmap of the serverset. We might need a mutex or a sync map here.